src/dcfldd_error.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * dcfldd_error.c -- replacement for non-portable GNU libc's error()
+ * Copyright (C) 2024 David da Silva Polverari
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "config.h"
```

Comment:
```suggestion
#if HAVE_CONFIG_H
#  include "config.h"
#endif
```
(e.g. for consistency with at least some other files)

Reply:
We are using autotools, so I think we don't need to make it conditional as `config.h` is always there (I think? maybe I'm wrong, going to have a look).

I think other files have those conditional preprocessor directives due to historical reasons. Anyways, it is something we could tackle in other PRs, what do you think?

Reply:
@davidpolverari I'm good with not adding the guard and dropping all the other guards in a dedicated pull request.